src/acquisition/covid_hosp/common/utils.py

```diff
@@ -8,219 +8,211 @@
-      if len(value) > length:
-        raise CovidHospException(f"Value '{value}':{len(value)} longer than max {length}")
-      return value
-    return limited_string
-
-  GEOCODE_LENGTH = 32
-  GEOCODE_PATTERN = re.compile(r'POINT \((-?[0-9.]+) (-?[0-9.]+)\)')
-  def limited_geocode(value):
-    if len(value) < Utils.GEOCODE_LENGTH:
-      return value
-    # otherwise parse and set precision to 6 decimal places
-    m = Utils.GEOCODE_PATTERN.match(value)
-    if not m:
-      raise CovidHospException(f"Couldn't parse geocode '{value}'")
-    return f'POINT ({" ".join(f"{float(x):.6f}" for x in m.groups())})'
-
-  def issues_to_fetch(metadata, newer_than, older_than, logger=False):
-    """
-    Construct all issue dates and URLs to be ingested based on metadata.
-
-    Parameters
-    ----------
-    metadata pd.DataFrame
-      HHS metadata indexed by issue date and with column "Archive Link"
-    newer_than Date
-      Lower bound (exclusive) of days to get issues for.
-    older_than Date
-      Upper bound (exclusive) of days to get issues for
-    logger structlog.Logger [optional; default False]
-      Logger to receive messages
-    Returns
-    -------
-      Dictionary of {issue day: list of (download urls, index)}
-      for issues after newer_than and before older_than
-    """
-    daily_issues = {}
-    n_beyond = 0
-    n_selected = 0
-    for index in sorted(set(metadata.index)):
-      day = index.date()
-      if day > newer_than and day < older_than:
-        urls = metadata.loc[index, "Archive Link"]
-        urls_list = [(urls, index)] if isinstance(urls, str) else [(url, index) for url in urls]
-        if day not in daily_issues:
-          daily_issues[day] = urls_list
-        else:
-          daily_issues[day] += urls_list
-        n_selected += len(urls_list)
-      elif day >= older_than:
-        n_beyond += 1
-    if logger:
-      if n_beyond > 0:
-        logger.info("issues available beyond selection", on_or_newer=older_than, count=n_beyond)
-      logger.info("issues selected", newer_than=str(newer_than), older_than=str(older_than), count=n_selected)
-    return daily_issues
-
-  @staticmethod
-  def merge_by_key_cols(dfs, key_cols, logger=False):
-    """Merge a list of data frames as a series of updates.
-
-    Parameters:
-    -----------
-      dfs : list(pd.DataFrame)
-        Data frames to merge, ordered from earliest to latest.
-      key_cols: list(str)
-        Columns to use as the index.
-      logger structlog.Logger [optional; default False]
-        Logger to receive messages
-
-    Returns a single data frame containing the most recent data for each state+date.
-    """
-
-    dfs = [df.set_index(key_cols) for df in dfs
-           if not all(k in df.index.names for k in key_cols)]
-    result = dfs[0]
-    if logger and len(dfs) > 7:
-      logger.warning(
-        "expensive operation",
-        msg="concatenating more than 7 files may result in long running times",
-        count=len(dfs))
```

Comment:
black's line-wrap behavior is basically to try to reformat until lines have a line-length less than its `line-length` setting (set in `pyproject.toml` in this repo to 200 by Sam). see [here for Black's docs on it](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length).

so the inconsistency you're seeing is because it's only reformatting lines that are over 200 long.

i'm personally fine with having such a long line setting, but i know that it breaks from the programming standard of 80. even black's default setting is 100-ish. 

the difficulty with setting black to short line-lengths is that it is very likely to break semantic spacing in that case, like in the example below
```
Columndef("previous_day_admission_adult_covid_confirmed_18-19_7_day_sum", "previous_day_admission_adult_covid_confirmed_18_19_7_day_sum", int),
Columndef("previous_day_admission_adult_covid_confirmed_20-29_7_day_sum", "previous_day_admission_adult_covid_confirmed_20_29_7_day_sum", int),
```

if we had set the default line-length shorter, all of these would like 

```
Columndef(
    "previous_day_admission_adult_covid_confirmed_18-19_7_day_sum",
    "previous_day_admission_adult_covid_confirmed_18_19_7_day_sum", 
    int
),
Columndef(
    "previous_day_admission_adult_covid_confirmed_20-29_7_day_sum",
    "previous_day_admission_adult_covid_confirmed_20_29_7_day_sum", 
    int
),
```

i think both are acceptable and have tradeoffs (e.g. vertical space conservation, alignment, horizontal space conservation, consistency of formatting).

not sure how to arrive at a consensus in this space, where prefs will vary a lot.

Reply:
> so the inconsistency you're seeing is because it's only reformatting expressions with lines that are over 200 long.

Gotcha, didn't realize that the line-length setting was longer than the usual 80 characters.

If we're going to be running all of our repos through `black`, it would be nice to have them formatted with the same settings, so maybe it's worth messing around with those a bit here? Or do you thing the 200-length line is already as good as it gets for automated formatting?

Are we turning on automated linting after we finish this pass through `black`? Do we need to consider how `black`'s settings will interact with that (e.g. trying to minimize errors we'll need to fix manually)?

> not sure how to arrive at a consensus in this space, where prefs will vary a lot.

I want to emphasize that my preferences here are very weak. As long as it's reasonable/readable, it's fine.

Reply:
Let me give the 100 line-length a shot in a separate PR and see how different it looks

Reply:
https://github.com/cmu-delphi/delphi-epidata/pull/1189